src/utils/tape-ava-helpers.js

```diff
@@ -115,6 +115,49 @@ export function rewriteAssertionsAndTestArgument(j, path) {
 }
 
 /**
+ * Rewrite test callback to be able to destructure its argument
+ *
+ * test(({ok}) => {ok()}) to test(t => {ok()})
+ */
+export function rewriteDestructuredTArgument(fileInfo, j, ast, testFunctionName) {
+    const rewriteOneArgument = p => {
+        // The last arg is the test callback
+        const lastArg = p.value.arguments[p.value.arguments.length - 1];
+        if (lastArg && lastArg.params && lastArg.params[0]
+                && lastArg.params[0].type === 'ObjectPattern') {
+            const objectPattern = lastArg.params[0];
+            const keys = objectPattern.properties.map(prop => prop.key.name);
+            lastArg.params[0] = j.identifier('t');
+
+            keys.forEach(key => {
+                j(lastArg).find(j.CallExpression, {
+                    callee: { name: key },
+                }).forEach(assertion => {
+                    j(assertion).replaceWith(j.callExpression(
+                        j.memberExpression(
+                            j.identifier('t'),
+                            j.identifier(key),
+                        ),
+                        assertion.node.arguments,
+                    ));
+                });
+            });
+        }
+    };
+    ast.find(j.CallExpression, {
```

Comment:
I'm not sure what the two cases (`callee.object.name` case and the `callee.name`) is. Can you add a comment? And ensure both cases are hit by your tests.

Consider doing a filter instead of searching twice. Not sure if it is more readable:
```javascript
ast.find(j.CallExpression)
   .filter({ node } => node.callee.name === testFunctionName || node.callee.object.name === testFunctionName)
   .forEach(rewriteOneArgument);
```

( not sure if the syntax is correct, coded this on GitHub 😃  )